reframe/core/pipeline.py

```diff
@@ -707,7 +707,7 @@ def __new__(cls, *args, **kwargs):
         obj = super().__new__(cls)
 
         # Set the test parameters in the object
-        cls._set_param_space(obj)
+        cls._construct_params(obj)
```

Comment:
I think after all `_init_params()` is a better name, because we are kind of supplementing the constructor here and in Python that's the `__init__()`.

Reply:
Agree. That name makes a lot more sense.